pages/docs/tutorials/javascript/setting-up.md

```diff
@@ -1,106 +1,154 @@
-            }
-        }
-
-        val jsTest by getting {
-            dependencies {
-                implementation(kotlin("test-js"))
-            }
+2. Change the code of `styledDiv` to show your name backwards.  
+   
+    * Use the standard library function `reversed()` to reverse your name.
+    * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) for your reversed 
+    name by adding a dollar sign `$` and enclosing it in curly braces – `${state.name.reversed()}`.
+    
+    <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+    
+    ```kotlin
+    styledDiv {
+        css {
+            +WelcomeStyles.textContainer
         }
+        +"Hello ${state.name}!"
+        +" Your name backwards is ${state.name.reversed()}!"
     }
-}
-```
-</div>
+    ```
+    
+    </div>
+
+3. Save your changes to the file.
+
+4. Go to the browser and enjoy the result.  
+    You will see the changes only if your previous application is still running. If you've stopped your application, [run it again](#run-the-application).
+   
+<img class="img-responsive" src="{{ url_for('tutorial_img', filename='javascript/setting-up/js-output-2.png') }}" alt="Web browser with a reversed name" />
+
+### Add an image
+
+1. Open the file `welcome.kt` in `src/main/kotlin`.  
+
+2. Add a `div` container with a child image element `img` after the `styledInput` block.  
+   
+    > Make sure that you import the `react.dom.*` and `styled.*` packages.
+    {:.note}       
+    
+    <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+    
+    ```kotlin
+    div {
+        img(src = "https://placekitten.com/408/287") {}
+    }
+    ```
+    
+    </div>
+                                                                                                                                                                                                                                                         
+3. Save your changes to the file.
+
+4. Go to the browser and enjoy the result.  
+    You will only see the changes if your previous application is still running. If you've stopped your application, [run it again](#run-the-application).
+   
+<img class="img-responsive" src="{{ url_for('tutorial_img', filename='javascript/setting-up/js-output-3.png') }}" alt="Web page with with an image" width="500"/>
+
+### Add a button that changes text
+
+1. Open the file `welcome.kt` in `src/main/kotlin`.  
+
+2. Add a `button` element with an `onClickFunction` event.  
+   
+    > Make sure that you import the package `kotlinx.html.js.*`.
+    {:.note}       
+    
+    <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+    
+    ```kotlin
+    button {
+        attrs.onClickFunction = {
+            setState(
+                WelcomeState(name = "Some name")
+            )
+        }
+        +"Change name"
+    }   
```

Comment:
This line also for some reason renders an extra space, and indents the first layer with 5 instead of 4 spaces:
![image](https://user-images.githubusercontent.com/2178959/88584771-7827ee80-d052-11ea-927e-5cbc7423ef5d.png)

Reply:
🛠